sympy/series/tests/test_limits.py

```diff
@@ -96,6 +96,11 @@ def test_basic1():
     assert limit(log(1/x) + 1/sin(x), x, 0, dir = '+') == oo
     assert limit(log(1/x) + 1/x, x, 0, dir = '+') == oo
 
+    # test acoth function at 0
+    assert limit(acoth(x), x, 0, '+') == -I*pi/2
+    assert limit(acoth(x), x, 0, '-') == I*pi/2
```

Comment:
There should be more tests for cases like`acoth(-x)` and `acoth(x - I*x)` as `x -> 0`.

Reply:
True , these are significant tests , will look into them . You could also give your thoughts regarding the comments above !
https://github.com/sympy/sympy/pull/22995#issuecomment-1028648751 this one specifically!

Reply:
I have updated the code to answer all possible cases and added test for the same, things which work now
```
>>> limit(acosh(x), x, 0)
I*pi/2
>>> limit(acoth(x), x, 0)
-I*pi/2
>>> limit(acoth(x), x, 0, '-')
I*pi/2
>>> limit(acoth(-x), x, 0)
I*pi/2
>>> limit(acoth(-x), x, 0, '-')
-I*pi/2
>>> limit(acoth(x**2), x, 0)
-I*pi/2
>>> limit(acoth(x**2), x, 0, '-')
-I*pi/2
>>> limit(acoth(I*x), x, 0)
-I*pi/2
>>> limit(acoth(I*x), x, 0, '-')
I*pi/2
>>> limit(acoth(I*x**2), x, 0)
-I*pi/2
>>> limit(acoth(I*x**2), x, 0, '-')
-I*pi/2
>>>
>>>
>>> limit(acoth(x - I*x), x, 0)
I*pi/2
>>> limit(acoth(x - I*x), x, 0, '-')
-I*pi/2
>>> limit(acoth(x**2 - I*x**2), x, 0)
I*pi/2
>>> limit(acoth(x**2 - I*x**2), x, 0, '-')
I*pi/2
```